portfolio/src/main/webapp/index.html

```diff
@@ -2,17 +2,216 @@
+    <div class="numbertext">4 / 10</div>
+    <img src="image/jerusalem.jpg" style="width:100%">
+    <div class="text">My trip to Jerusalem</div>
+  </div>
+
+  <div class="mySlides fade">
+    <div class="numbertext">5 / 10</div>
+    <img src="image/thailand.png" style="width:100%">
+    <div class="text">Kanchanaburi Thailand, the River Kwai</div>
+  </div>
+
+  <div class="mySlides fade">
+    <div class="numbertext">6 / 10</div>
+    <img src="image/bigbuddha.png" style="width:100%">
+    <div class="text">The Big Buddha at Ngong Ping</div>
+  </div>
+
+  <div class="mySlides fade">
+    <div class="numbertext">7 / 10</div>
+    <img src="image/hkfishingtown.jpg" style="width:100%">
+    <div class="text">A small fishing town at the base of Ngong Ping</div>
+  </div>
+
+  <div class="mySlides fade">
+    <div class="numbertext">8 / 10</div>
+    <img src="image/canada.jpg" style="width:100%">
+    <div class="text">Canada heading to the mine</div>
+  </div>
+
+  <div class="mySlides fade">
+    <div class="numbertext">9 / 10</div>
+    <img src="image/govball.png" style="width:100%">
+    <div class="text">Governer's Ball Music Festival</div>
+  </div>
+
+  <div class="mySlides fade">
+    <div class="numbertext">10 / 10</div>
+    <img src="image/towerbridge.jpg" style="width:100%">
+    <div class="text">Tower Bridge with my Girlfriend Ashley, on our first trip to London</div>
+  </div>
+
+  <!-- Next and previous buttons -->
+  <a class="prev" onclick="plusSlides(-1)">&#10094;</a>
+  <a class="next" onclick="plusSlides(1)">&#10095;</a>
+</div>
+<br>
+
+<!-- The dots/circles -->
+<div style="text-align:center">
+  <span class="dot" onclick="currentSlide(1)"></span>
+  <span class="dot" onclick="currentSlide(2)"></span>
+  <span class="dot" onclick="currentSlide(3)"></span>
+  <span class="dot" onclick="currentSlide(4)"></span>
+  <span class="dot" onclick="currentSlide(5)"></span>
+  <span class="dot" onclick="currentSlide(6)"></span>
+  <span class="dot" onclick="currentSlide(7)"></span>
+  <span class="dot" onclick="currentSlide(8)"></span>
+  <span class="dot" onclick="currentSlide(9)"></span>
+  <span class="dot" onclick="currentSlide(10)"></span>
+
+    <script>
+let slideIndex = 0;
+showSlidesAuto();
+
+// Next/previous controls
+function plusSlides(n) {
+  showSlides(slideIndex += n);
+}
+
+// Thumbnail image controls
+function currentSlide(n) {
+  showSlides(slideIndex = n);
+}
+
+function showSlides() {
+  let i;
+  const slides = document.getElementsByClassName("mySlides");
+  const dots = document.getElementsByClassName("dot");
+  for (i = 0; i < slides.length; i++) {
+    slides[i].style.display = "none";  
+  }
+  if (slideIndex > slides.length) {slideIndex = 1}    
+  for (i = 0; i < dots.length; i++) {
+    dots[i].className = dots[i].className.replace(" active", "");
+  }
+  slides[slideIndex-1].style.display = "block";  
+  dots[slideIndex-1].className += " active";
+}
+function showSlidesAuto() {
```

Comment:
This function looks _really_ similar to `showSlides()`. Duplicate code takes a lot more work to maintain because every change to `showSlides` needs to happen in `showSlidesAuto` otherwise you'll end up with really painful bugs.

In this case, what if `showSlidesAuto` was something like:

```javascript
function showSlidesAuto() {
  slideIndex++;
  showSlides()
}
setTimeout(showSlidesAuto, 10000);
```

Reply:
took this suggestion, and refactored the i variable out too